.github/workflows/pkgci_build_packages.yml

```diff
@@ -30,6 +30,15 @@ jobs:
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           submodules: true
+      - name: Setup Python
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.3
+        with:
+          python-version: 3.12
+          cache: "pip"
+      - name: Install Python packages
+        run: |
+          pip install packaging
+          pip freeze
       - name: Write version info
```

Comment:
Did you want pkgci to be using the new version scheme in this PR? I think you'll also need to update https://github.com/iree-org/iree/blob/8158a8cca1c1e812b9a7a8892a1ab15b9d6edb01/.github/workflows/pkgci.yml#L32-L38

The logs on this PR https://github.com/iree-org/iree/actions/runs/11693249175/job/32570265367?pr=19034#step:6:41

still have
```
Run cat << EOF > version_info.json
/home/runner/_work/iree/iree/version_info.json
{
  "package-suffix": "",
  "package-version": "0.dev1",
  "iree-revision": "d72624421b86fd4b4f91ae0c612566eca1023980"
}
```

thanks to https://github.com/iree-org/iree/actions/runs/11693249175/job/32570265367?pr=19034#step:1:20
```
 Inputs
    package_version: 0.dev1
```

Reply:
Yeah, I was unsure who to handle this. `0.dev1` is what is written to the `version_info.json` and thus only applies to packages like `iree-tools-tf` and `iree-tools-tflite`. For `iree-compiler` and `iree-runtime`, you'd get `3.0.0.dev1` (setuptools normalize the version number, the logs show `3.0.00.dev1`). I added some more logging (see e.g. https://github.com/iree-org/iree/actions/runs/11702485462/job/32590788790?pr=19034#step:6:1) but I am happy to switch e.g. to `3.0.0.dev+${githash}` for `iree-compiler` and `iree-runtime`. WDYT?

Reply:
Thanks for the logging!

> but I am happy to switch e.g. to `3.0.0.dev+${githash}` for `iree-compiler` and `iree-runtime`. WDYT?

Yes please. The extra `0` in `3.0.00.dev1` is odd, and we should get the PR number of git hash in the suffix.

Reply:
Oh seems I misread, I just pushed a commit that gets you `X.Y.Z.dev+${githash}`. Should we go with `X.Y.Z.dev${{ github.run_number }}+${githash}` instead?

Reply:
We synced offline and agreed on `X.Y.Z.dev+${githash}`.